src/state/medicaltypes/actions.ts

```diff
@@ -0,0 +1,198 @@
+import isEmpty from "lodash.isempty";
+import { Dispatch } from "redux";
+import {
+  Configuration,
+  MedicalTypeControllerApi,
+  MedicalTypeDTO,
+  CreateMedicalTypeUsingPOSTRequest,
+  UpdateMedicalTypeUsingPUTRequest,
+  OperationOpts,
+} from "../../generated";
+import { applyTokenMiddleware } from "../../libraries/apiUtils/applyTokenMiddleware";
+import { IAction } from "../types";
+import {
+  GET_MEDICALTYPE_FAIL,
+  GET_MEDICALTYPE_LOADING,
+  GET_MEDICALTYPE_SUCCESS,
+  CREATE_MEDICALTYPE_FAIL,
+  CREATE_MEDICALTYPE_LOADING,
+  CREATE_MEDICALTYPE_RESET,
+  CREATE_MEDICALTYPE_SUCCESS,
+  EDIT_MEDICALTYPE_LOADING,
+  EDIT_MEDICALTYPE_FAIL,
+  EDIT_MEDICALTYPE_RESET,
+  EDIT_MEDICALTYPE_SUCCESS,
+  DELETE_MEDICALTYPE_LOADING,
+  DELETE_MEDICALTYPE_FAIL,
+  DELETE_MEDICALTYPE_RESET,
+  DELETE_MEDICALTYPE_SUCCESS,
+} from "./consts";
+
+const medicalTypeControllerApi = new MedicalTypeControllerApi(
+  new Configuration({ middleware: [applyTokenMiddleware] })
+);
+
+export const newMedicalType =
+  ( medicalType: MedicalTypeDTO) => //, ignSimilar: boolean
+  (dispatch: Dispatch<IAction<null, {}>>): void => {
+    dispatch({
+      type: CREATE_MEDICALTYPE_LOADING,
+    });
+
+    var newMedicalTypeRequest: CreateMedicalTypeUsingPOSTRequest = { medicalTypeDTO: medicalType }
+    medicalTypeControllerApi.createMedicalTypeUsingPOST(newMedicalTypeRequest).subscribe( 
+      () => {
+        dispatch({
+          type: CREATE_MEDICALTYPE_SUCCESS,
+        });
+      },
+      (error) => {
+        dispatch({
+          type: CREATE_MEDICALTYPE_FAIL,
+          error,
+        });
+      }
+    );
+  };
+
+  export const createMedicalReset =
+  () =>
+  (dispatch: Dispatch<IAction<null, {}>>): void => {
+    dispatch({
+      type: CREATE_MEDICALTYPE_RESET,
+    });
+  };
+
+export const editMedicalType =
+  ( updateMedicalType: MedicalTypeDTO) =>
+  (dispatch: Dispatch<IAction<null, {}>>): void => {
+    dispatch({
+      type: EDIT_MEDICALTYPE_LOADING,
+    });
+
+    var updateMedicalTypeRequest: UpdateMedicalTypeUsingPUTRequest = { medicalTypeDTO: updateMedicalType }
+    medicalTypeControllerApi
+      .updateMedicalTypeUsingPUT(updateMedicalTypeRequest)
```

Comment:
We can directly use ` medicalTypeControllerApi
      .updateMedicalTypeUsingPUT({ medicalTypeDTO: updateMedicalType })` and remove the variable declaration. May I am wrong please, is there a particular reason to create variable?